Implement dark mode

tuhalang opened the pull request:
Resolve #399 
- Create switch button on setting page
- Switch light mode or dark mode when clicking the switch button

tuhalang committed:
Implement dark mode

tuhalang committed:
fix go lint

Sirmorrison commented on ui/decredmaterial/label.go:
```diff
@@ -57,6 +59,15 @@ func (t *Theme) Label(size unit.Value, txt string) Label {
+	l.Color = color
+	return l
+}
+
```
I dont see the need for this codes as the Label widget already has a color parameter than can be set by calling `Label.Color`

Sirmorrison commented on ui/compontents.go:
```diff
@@ -29,10 +29,10 @@ func (page pageCommon) layoutBalance(gtx layout.Context, amount string) layout.D
 	return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Baseline}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
-			return page.theme.Label(values.TextSize20, mainText).Layout(gtx)
+			return page.theme.LabelColor(values.TextSize20, mainText, page.theme.Color.DeepBlue).Layout(gtx)
```
```suggestion
			label := return page.theme.Label(values.TextSize20, mainText)
			label.Color = page.theme.Color.DeepBlue
  			return label.Layout(gtx)
```

Sirmorrison commented on ui/compontents.go:
```diff
@@ -29,10 +29,10 @@ func (page pageCommon) layoutBalance(gtx layout.Context, amount string) layout.D
 		}),
 		layout.Rigid(func(gtx C) D {
-			return page.theme.Label(values.TextSize14, subText).Layout(gtx)
+			return page.theme.LabelColor(values.TextSize14, subText, page.theme.Color.DeepBlue).Layout(gtx)
```
same as above

Sirmorrison commented on ui/compontents.go:
```diff
@@ -65,7 +65,7 @@ func (page pageCommon) layoutUSDBalance(gtx layout.Context) layout.Dimensions {
 						return padding.Layout(gtx, func(gtx C) D {
 							amountDCRtoUSDString := formatUSDBalance(page.printer, page.amountDCRtoUSD)
-							return page.theme.Label(values.TextSize14, amountDCRtoUSDString).Layout(gtx)
+							return page.theme.LabelColor(values.TextSize14, amountDCRtoUSDString, page.theme.Color.DeepBlue).Layout(gtx)
```
Instead of using Label, your code can best be...
```suggestion
							 return page.theme.Body2(amountDCRtoUSDString).Layout(gtx)
```

Also, you dont need the LabelColor() method.

tuhalang committed:
Fix feedback:
- Remove function LableColor
- Add dark mode for ticket page

tuhalang committed:
Fix crash app when startup
- Sometimes, the application is crashed because the wallet hasn't loaded
- Checking the wallet is not nil before read the config properties

tuhalang committed:
Fix feedback:
- Fix darkmode for modals
- Fix darkmode for receive address

beansgum reviewed (changes requested):
Some texts are not readable 

![Screenshot 2021-05-21 at 08 07 18](https://user-images.githubusercontent.com/19960200/119097185-58f8b400-ba0c-11eb-852f-70c923a34859.png)
![Screenshot 2021-05-21 at 08 08 13](https://user-images.githubusercontent.com/19960200/119097190-5a29e100-ba0c-11eb-91ef-1a64d24c3a6f.png)
<img src="https://user-images.githubusercontent.com/19960200/119097198-5bf3a480-ba0c-11eb-806f-e46b2efd2f3d.png" width="300">
<img src="https://user-images.githubusercontent.com/19960200/119097199-5d24d180-ba0c-11eb-986a-6ba0160c095d.png" width="300">

tuhalang committed:
Fix feedback:
- Fix some texts are not readable when switch darkmode

oshorefueled commented on ui/overview_page.go:
```diff
@@ -162,7 +163,7 @@ func (win *Window) OverviewPage(c pageCommon) layout.Widget {
 
 	return func(gtx C) D {
-		pg.Handler(gtx, c)
+		pg.Handler(gtx, c, win)
```
You shouldn't need to pass the window struct to the handler method.  It seems you're trying to call the reload method from the handler after toggling the dark mode option. Is there a reason you need to re-initialize the icons and pages after toggling the dark mode option?

tuhalang replied:
Yes, i need re-initialize pages after toggling the dark mode option.

oshorefueled commented on ui/page.go:
```diff
@@ -377,6 +382,10 @@ func (win *Window) addPages(decredIcons map[string]image.Image) {
 
+func (win *Window) reloadPage(common pageCommon) {
+	win.loadPage(common.icons)
+}
```
This seems like a redundant wrapper for the loadPage method.

oshorefueled commented on ui/overview_page.go:
```diff
@@ -639,7 +640,16 @@ func (pg *overviewPage) walletSyncBox(gtx layout.Context, inset layout.Inset, de
+	if win.wallet != nil {
+		isDarkModeOn := win.wallet.ReadBoolConfigValueForKey("isDarkModeOn")
+		if isDarkModeOn != win.theme.DarkMode {
+			win.theme.SwitchDarkMode(isDarkModeOn)
```
Why is this method called in the overview and settings page. Shouldn't it be called just once from the settings page?

tuhalang replied:
I called it in the settings page to handle the button click event dark mode. In the overview page because it is the first page loaded when app startup, i can read config from the wallet in this page and switch to the dark mode.

oshorefueled commented on ui/compontents.go:
```diff
@@ -71,7 +79,8 @@ func (page pageCommon) layoutUSDBalance(gtx layout.Context) layout.Dimensions {
 							amountDCRtoUSDString := formatUSDBalance(page.printer, page.amountDCRtoUSD)
-							return page.theme.Label(values.TextSize14, amountDCRtoUSDString).Layout(gtx)
+							return page.theme.Body2(amountDCRtoUSDString).Layout(gtx)
+							//return page.theme.LabelColor(values.TextSize14, amountDCRtoUSDString, page.theme.Color.DeepBlue).Layout(gtx)
```
remove redundant comments

oshorefueled commented on ui/compontents.go:
```diff
@@ -30,15 +30,23 @@ const (
 	return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Baseline}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
-			return page.theme.Label(values.TextSize20, mainText).Layout(gtx)
+			label := page.theme.Label(values.TextSize20, mainText)
+			if isSwitchColor {
+				label.Color = page.theme.Color.DeepBlue
+			}
+			return label.Layout(gtx)
```
Why is the colour for the text being done specifically here, shouldn't it be affected like other text colours in the app?

tuhalang replied:
Because the text in here alway display black color independent dark or light mode
![image](https://user-images.githubusercontent.com/33905113/119283394-11219900-bc67-11eb-93ad-24714a33e181.png)

tuhalang committed:
Remove redundant comments and function
